runtime/src/main/java/org/corfudb/runtime/LiteRoutingQueueListener.java

```diff
@@ -72,105 +77,85 @@ public LiteRoutingQueueListener(CorfuStore corfuStore, String sourceSiteId, Stri
+            Preconditions.checkState(entry.getOperation() == CorfuStreamEntry.OperationType.UPDATE);
             Queue.CorfuGuidMsg key = (Queue.CorfuGuidMsg) entry.getKey();
             Queue.RoutingTableEntryMsg msg = (Queue.RoutingTableEntryMsg) entry.getPayload();
+            allRQMsgs.add(msg);
+
             Queue.CorfuQueueMetadataMsg metadataMsg = (Queue.CorfuQueueMetadataMsg) entry.getMetadata();
             allEntries.add(new Table.CorfuQueueRecord(key, metadataMsg, msg));
-            if (currentType.equals(Queue.ReplicationType.LAST_FULL_SYNC_ENTRY)) {
-                currentType = msg.getReplicationType();
-            } else if (!currentType.equals(msg.getReplicationType())) {
-                throw new LogReplicationClientException("Not expecting mixed event types "+ currentType +" vs "
-                    +msg.getReplicationType());
-            }
+            currentReplicationType = msg.getReplicationType();
         }
         if (allEntries.isEmpty()) {
             return;
         }
-        boolean entriesProcessed;
+        int numEntriesProcessed;
         long now = System.currentTimeMillis();
-        long whenQrecordWasCreated = CorfuGuid.getTimestampFromGuid(allEntries.get(0).getRecordId().getInstanceId(), now);
-        if (currentType.equals(Queue.ReplicationType.LOG_ENTRY_SYNC)) {
-            log.info("LRQRecvListener delivering {} LOG_ENTRY updates took {}ms end to end",
-                    allEntries.size(), now - whenQrecordWasCreated);
-            entriesProcessed = processUpdatesInLogEntrySync(allEntries.stream().sorted().map(q ->
-                    (Queue.RoutingTableEntryMsg)q.getEntry()).collect(Collectors.toList()));
-        } else {
-            log.info("LRQRecvListener delivering {} FULLSYNC updates took {}ms end to end",
-                    allEntries.size(), now - whenQrecordWasCreated);
-            entriesProcessed = processUpdatesInSnapshotSync(allEntries.stream().sorted().map(q ->
-                    (Queue.RoutingTableEntryMsg)q.getEntry()).collect(Collectors.toList()));
-        }
+        long whenQRecordWasCreated = CorfuGuid.getTimestampFromGuid(allEntries.get(0).getRecordId().getInstanceId(),
+            now);
 
-        if (entriesProcessed) {
+        log.info("Delivering {} {} updates took {}ms end to end", allEntries.size(), currentReplicationType,
+            now - whenQRecordWasCreated);
+        numEntriesProcessed = processBatch(currentReplicationType, allRQMsgs);
+
+        if (numEntriesProcessed == allRQMsgs.size()) {
             try (TxnContext txnContext = corfuStore.txn(CORFU_SYSTEM_NAMESPACE)) {
                 List<UUID> noStreamTags = Collections.emptyList();
-                allEntries.forEach(q -> txnContext.logUpdateDelete(recvQ, q.getRecordId(), noStreamTags, corfuStore)
-                );
+                allEntries.forEach(q -> txnContext.logUpdateDelete(recvQ, q.getRecordId(), noStreamTags, corfuStore));
```

Comment:
nit: we can pull out the "delete" logic from here and performFullSync to a new method, and call that method here